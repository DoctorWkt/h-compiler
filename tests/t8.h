char a[]= "This is a string";

void main()
{
  char *x;

  // Print the string
  x=a;
  while (*x != 0) {
    putchar(*x); x++;
  }
  putchar(10);

  // Shorten it
  a[4]= 0;

  // Print the string
  x=a;
  while (*x != 0) {
    putchar(*x); x++;
  }
  putchar(10);
}

