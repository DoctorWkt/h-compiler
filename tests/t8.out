This is a string
This
